### ISSUE TITLE
[BUG] Cannot undo rescheduling

### ISSUE THREAD

**Issue by topherbuckley** [open-spaced-repetition/fsrs4anki#383]
**Describe the bug**
When using the **FSRS4Anki Helper --> Reschedule All Cards** followed by **FSRS4Anki Helper --> Undo rescheduling in all cards** my collection is still vastly different from where it started. 

**To Reproduce**
1. Open collection in Anki desktop (Ubuntu 22.04)
2. Open Stats and show full collection for 1 year and include backlog. 
3. Print to pdf
4. **FSRS4Anki Helper --> Reschedule All Cards** 
5. repeat steps 2-3
6. **FSRS4Anki Helper --> Undo rescheduling in all cards** 
7. repeat steps 2-3

**Expected behavior**
I would expect the undo rescheduling to bring my collection and scheduling back to where it was prior to rescheduling. 

**Screenshots**
From left to right. Step 3, Step 5, Step 7 from list of steps above. 
![image](https://github.com/open-spaced-repetition/fsrs4anki/assets/15166572/b6901a03-c012-4bca-9661-27d88c4ea222)

**Environment**
 - Anki version: 2.1.65 (82caffec)⁩
   Python 3.9.15 Qt 6.5.0 PyQt 6.5.0
 - OS: Ubuntu 22.04
 - Scheduler version: Not sure where to find all three off these individually, but I'm using everything from tag: v4.3.2 on this repo
 - Optimizer version: Installed from pip and also reads 4.3.2 according to `pip list`
 - Simulator version: Don't think I'm using this but correct me if I'm wrong.

**Comment by L-M-Sherlock** [comment]
Thanks for your report. Could you recover from the backup before you rescheduled all cards? It would be a bug need to fix. I want to reproduce the bug in my device.

**Comment by topherbuckley** [comment]
Yeah, I made a backup and turned off syncing before I tried this, so I recovered just fine. I'm happy to share my collection file if you'd like to try to reproduce it on a test profile. Where would be a good place to upload it for you? Without media its around 8MB. I'd prefer if my collection weren't public, but happy to email it or upload to drive/dropbox/etc.

**Comment by L-M-Sherlock** [comment]
My email address is jarrett.ye@outlook.com.

**Comment by topherbuckley** [comment]
Sent!
